### DISCUSSION TITLE
Does Ory Kratos work with multiple frontend selfservice flows?

### DISCUSSION THREAD

**Discussion by CMCDragonkai** [ory/kratos#1381] [Q&A] [answered]
I've been testing out the Ory Kratos system with this UI.

Based on the configuration of kratos.yaml, I've found that I have to hardcode the return UI urls and other urls to kratos.

Does that mean Kratos only supports a single selfservice flow? That is there cannot be multiple selfservice flow UIs in front of a single kratos system?

We were thinking that there would be different applications (with different UIs and different urls) that each would have their own login/registration flows that would all be managed by kratos. But this seems limited to a single UI.

**Comment by vinckr** [comment, marked as answer]
One easy fix for this is to use multiple Kratos deployments. They are lightweight enough to deploy once for each different frontend. 
You can have them share the same database or use a different database for each deployment, depending on your use case. 

There is other ways to achieve multi-tenancy (if that was where your question was headed) but they are not as easy to pull off.

**Reply by vinckr** [reply]
Hey @CMCDragonkai,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
